Update ReleaseChecklist.md: Add Homebrew formula bump description.

aarlt opened the pull request.

aarlt committed:
Update ReleaseChecklist.md: Add Homebrew formula bump description.

chriseth commented on ReleaseChecklist.md:
```diff
@@ -32,6 +32,10 @@
 ### Homebrew and MacOS
  - [ ] Update the version and the hash (``sha256sum solidity_x.x.x.tar.gz``) in https://github.com/ethereum/homebrew-ethereum/blob/master/solidity.rb
  - [ ] Take the binary from the ``b_osx`` run of the released commit in circle-ci and add it to the release page as ``solc-macos``.
+ - [ ] bump new homebrew formula 
```
is it possible to do this without the `brew` command?

aarlt replied:
we could write a script

chriseth replied:
What does it do?

aarlt replied:
maybe just checking out the last formula, changing it, git add, git commit, git push.. but i didnt looked into what `brew` is actually doing, maybe there is more magic involved

aarlt replied:
@ekpyron also found https://github.com/marketplace/actions/homebrew-bump-formula, maybe that would be easier to use

ekpyron replied:
Yeah... The action apparently was written by https://github.com/dawidd6 who appears to be homebrew maintainer, so that doesn't look bad - and we could just have it be triggered by the release announcement being published - so that could be rather convenient... but yeah, we might also get away with manually creating PRs bumping version and hashes (the brew command does some verification, maybe even a test build, but ultimately just created this PR: https://github.com/Homebrew/homebrew-core/pull/60549 - but the conversations @axic linked earlier sounded like the brew command is the preferred way they want people to use to create these PRs...).

ekpyron replied:
But yeah, I'd say let's first just watch that PR and if and how fast it gets merged and then decide how to proceed and do things in the future.

aarlt replied:
The PR was merged, see https://github.com/Homebrew/homebrew-core/commit/be38575ba77c11a55b9e77cf8c4ff4582c38f4c3, https://github.com/Homebrew/homebrew-core/commit/cfc449fdde19b61953777c10677c8b2f4d002d81. 

I think it would be nice to do this somehow automated, but if not: there seem to be linuxbrew docker images available https://github.com/Linuxbrew/docker. maybe we could do such PR's manually without needing macos. However, I tried `docker run -it linuxbrew/alpine` and it seem to be a bit buggy - it was just not simply working as expected ;) there where some strange errors, i didn't investigate.. but maybe after some research we could also use this

ekpyron replied:
For the record: it only took them around two hours to merge it I think. Also apparently they do automatically build the formulas to "bottles" (i.e. binaries), which is nice as well - the osx builds should be deterministic, shouldn't they? We could probably verify that the bottle hashes match our builds (in case we deem it necessary).

@aarlt I think the docker image is for "linux brew", i.e. the version of homebrew to be used as linux package manager - I'm not sure linux brew can be used for updating osx packages...

aarlt replied:
@ekpyron yes, you're right.. I thought that these recipes are platform independent.. but at least for the bottled part it should not work

ekpyron commented on ReleaseChecklist.md:
````diff
@@ -32,6 +32,10 @@
  - [ ] Take the binary from the ``b_osx`` run of the released commit in circle-ci and add it to the release page as ``solc-macos``.
+ - [ ] bump new homebrew formula 
+```
+brew bump-formula-pr --message="Update solidity to 0.7.1 with brew bump-formula-pr2" --url="https://github.com/ethereum/solidity/releases/download/v0.7.1/solidity_0.7.1.tar.gz" --sha256="c69205d902ac8dd66f5aeaa78a08e5411cdb26a0b9479fcf44f394b7008b484c" solidity
````
```suggestion
brew bump-formula-pr --message="Update solidity to 0.7.1 with brew bump-formula-pr" --url="https://github.com/ethereum/solidity/releases/download/v0.7.1/solidity_0.7.1.tar.gz" --sha256="c69205d902ac8dd66f5aeaa78a08e5411cdb26a0b9479fcf44f394b7008b484c" solidity
```
Not sure where this 2 was coming from :-). That's the message that ends up in the PR this creates apparently...

stackenbotten commented:
There was an error when running `chk_coding_style` for commit `0d63b65b6b8ed53fda3de95483fb464633ec67e2`:
```
Error: Trailing whitespace found:
 ReleaseChecklist.md:35: - [ ] bump new homebrew formula 

```
Please check that your changes are working as intended.

chriseth commented on ReleaseChecklist.md:
````diff
@@ -32,6 +32,10 @@
  - [ ] Update the version and the hash (``sha256sum solidity_x.x.x.tar.gz``) in https://github.com/ethereum/homebrew-ethereum/blob/master/solidity.rb
  - [ ] Take the binary from the ``b_osx`` run of the released commit in circle-ci and add it to the release page as ``solc-macos``.
+ - [ ] bump new homebrew formula
+```
````
Probably better to use placeholders for the version number and the hash.

Please also mention what has to be done to do this manualy.

chriseth commented:
@aarlt can you finish this, please?

stackenbotten commented:
There was an error when running `chk_coding_style` for commit `840af0d03eadaef8e07ec8b2e56dbdb59ef3d479`:
```
Error: Trailing whitespace found:
 ReleaseChecklist.md:35: - [ ] Bump homebrew formula 
 ReleaseChecklist.md:36: 

```
Please check that your changes are working as intended.

stackenbotten commented:
There was an error when running `chk_coding_style` for commit `92b2ed5f63dc1939cf22855d62287d8dbc98b79d`:
```
Error: Trailing whitespace found:
 ReleaseChecklist.md:36: 

```
Please check that your changes are working as intended.

aarlt commented:
@chriseth I added a description how to do this manually. I also saw that the used SPDX identifier is `deprecated`, so I tested the manual steps by updating the license (see https://github.com/Homebrew/homebrew-core/pull/61245). Basically `GPL-3.0` is not a valid SPDX identifier anymore, but `GPL-3.0-only` and `GPL-3.0-or-later` where introduced. For now I changed the license to `GPL-3.0-only`. Is that license correct?

aarlt commented:
@chriseth What would you say is the license that should be supplied in the formula? just `GPL-3.0-or-later` or something like `all_of: ["GPL-3.0-or-later", "MIT", "BSD-3-Clause", "Apache-2.0", "CC0-1.0"]` as suggested in https://github.com/Homebrew/homebrew-core/pull/61245?

chriseth commented:
I'm not sure what the meaning of `all_of` is, but "gpl-3.0 or later" is the right license for the bulk of the source.

chriseth commented on ReleaseChecklist.md:
```diff
@@ -32,6 +32,30 @@
+
+#### Bump homebrew formula manually
+
+- fork homebrew core repository `https://github.com/Homebrew/homebrew-core`
```
This is too much for the checklist. Isn't it almost exactly the same as in https://github.com/ethereum/solidity/pull/9734/files#diff-a64159d53d82ffc076743ededd94db5eR33 ? This CAN be done via the github web interface, can it?

aarlt replied:
:D yes, it is basically exactly the same as described there, somehow I missed that.. sure yes, it can be done via the github web interface.. i will simplify that

Rylan12 commented:
> I'm not sure what the meaning of `all_of` is, but "gpl-3.0 or later" is the right license for the bulk of the source.

Hi, Homebrew maintainer here.

The `all_of:` indicates that there are multiple licenses that need to apply. See [Homebrew's License Guidelines](https://docs.brew.sh/License-Guidelines) for more information about the syntax here.

Thus, the suggested line:

```ruby
license all_of: ["GPL-3.0-or-later", "MIT", "BSD-3-Clause", "Apache-2.0", "CC0-1.0"]
```

is equivalent to the SPDX identifier: `GPL-3.0-or-later AND MIT AND BSD-3-Clause AND Apache-2.0 AND CC0-1.0` (more info about SPDX license expressions [here](https://spdx.github.io/spdx-spec/appendix-IV-SPDX-license-expressions/))

It is possible that `GPL-3.0-or-later` is all that's needed. I only brought up the issue of the other licenses because I saw this statement in the [`README`](https://github.com/ethereum/solidity#license):

> Some third-party code has its [own licensing terms](https://github.com/ethereum/solidity/blob/develop/cmake/templates/license.h.in).

I got the specific licenses from [`cmake/templates/license.h.in`](https://github.com/ethereum/solidity/blob/develop/cmake/templates/license.h.in) (which is where the link directs).

Does this help to clear things up?

aarlt commented:
@Rylan12 Thank you for the clarification. We decided to go with `all_of`. I updated my PR.

chriseth commented:
Please squash and mention the wording in the PR description that causes this to be recognized as created by the script.

aarlt commented:
@chriseth I'm not sure I get what you mean. There is no reason that the PR need to be recognised as created by the script. A normal PR should work. The only thing may be that the commit message should be `solidity x.y.z`, if the formula was updated to version `x.y.z`.  But it seem to be  to a strict requirement.

chriseth commented:
If you make the PR look like it was generated by the utility, it gets merged quicker.
